napari/utils/action_manager.py

```diff
@@ -230,12 +231,21 @@ def bind_button(
                 f'{self._build_tooltip(name)} {extra_tooltip_text}'
             )
 
+        # buttons may hold references to layers, see
+        # https://github.com/napari/napari/issues/2027
+        ref_button = ref(button)
+        del button
+
         def _update_tt(event: ShortcutEvent):
             if event.name == name:
-                button.setToolTip(f'{event.tooltip} {extra_tooltip_text}')
+                button_ = ref_button()
+                # there is still a small chance for the ref to be None,
+                # but this to not be disconnected.
+                if button_ is not None:
+                    button_.setToolTip(f'{event.tooltip} {extra_tooltip_text}')
 
         # if it's a QPushbutton, we'll remove it when it gets destroyed
-        until = getattr(button, 'destroyed', None)
+        until = getattr(ref_button(), 'destroyed', None)
```

Comment:
@tlambert03 as far as I can tell the `until` here should disconnect `_update_tt` so `ref_button()` should never be None, though sometime it's the case. Can you see the reason why ? Or what I am missing?